### PR TITLE
Fix's the code editor

### PR DESCRIPTION
**Pull request by allancoding**
The code editor got broken and this fixes it.

### CODE REVIEW

File: public/js/editor.js [hunk 1]
```diff
@@ -1,9 +1,23 @@
 (function() {
     function editorMain() {
```
Comment (ethanaobrien) [inline public/js/editor.js:2]: What is stopping this function from being called multiple times?

File: public/js/editor.js [hunk 2]
```diff
@@ -193,6 +207,15 @@
     if (window.document) {
         editorMain();
     } else {
-        window.addEventListener("DOMContentLoaded", editorMain);
+        window.addEventListener("load", editorMain);
     }
+    var target = document.querySelector('title');
+    var observer = new MutationObserver(function(mutations) {
```
Comment (ethanaobrien) [inline public/js/editor.js:213]: Can we please use let or const instead of var

Comment (ethanaobrien) [inline public/js/editor.js:213]: What is this api? What does it do?

File: public/js/editor.js [hunk 3]
```diff
@@ -193,6 +207,15 @@
...
+        mutations.forEach(function(mutation) {
+            if (document.title != 'Code Editor · EmulatorJS' && window.location.pathname == '/editor') {
```
Comment (ethanaobrien) [inline public/js/editor.js:215]: This can be done differently string comparison can be messy, though we should always use type comparison

**Review by ethanaobrien — CHANGES_REQUESTED** [review]

File: public/js/editor.js (hunk 2 above)
Comment (allancoding) [reply public/js/editor.js:213]: This fix's a bug with nuxtjs

File: public/js/editor.js (hunk 3 above)
Comment (allancoding) [reply public/js/editor.js:215]: I changed it to this: document.title !== 'Code Editor · EmulatorJS' && window.location.pathname === '/editor'.

File: public/js/editor.js (hunk 1 above)
Comment (allancoding) [reply public/js/editor.js:2]: loading the system.js file in a SPA website

File: public/js/editor.js [hunk 4]
```diff
@@ -1,9 +1,23 @@
...
         if (!window.systems) {
-            document.getElementById('systemJS').addEventListener('load', loadsystems);
+            try {
+                document.getElementById('systemJS').addEventListener('load', loadsystems);
+            } catch(e) {
+                setTimeout(function(){
+                    editorMain();
+                }, 100);
+                return;
+            }
```
[the remark this answers is not in the document]
Comment (allancoding) [reply public/js/editor.js:11]: The way pages are loaded it a SPA prevent the use of event listeners for when pages are done loading

File: public/js/editor.js [hunk 5]
```diff
@@ -1,9 +1,23 @@
...
+        }
+        try {
+            document.getElementById('start').addEventListener('click', loadsystems);
+        } catch(e) {
+            setTimeout(function(){
+                editorMain();
+            }, 100);
+            return;
```
[the remark this answers is not in the document]
Comment (allancoding) [reply public/js/editor.js:19]: I will merge them

File: public/js/editor.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (ethanaobrien) [reply public/js/editor.js:11]: What if this is a slower device? Anything involving a timer and something loading should never be done

File: public/js/editor.js (hunk 2 above)
Comment (ethanaobrien) [reply public/js/editor.js:213]: What bug?

File: public/js/editor.js (hunk 3 above)
Comment (ethanaobrien) [reply public/js/editor.js:215]: But why are we comparing strings at all? Why are we not moving it to the init function with the rest of the init code?

File: public/js/editor.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (allancoding) [reply public/js/editor.js:11]: The timer is there so that it wont hit the max callback number

[the remark this answers is not in the document]
Comment (ethanaobrien) [reply public/js/editor.js:11]: What do you mean? Why are we not initializing this listener on domcontentload?

[the remark this answers is not in the document]
Comment (allancoding) [reply public/js/editor.js:11]: That listener only runs on first load, the page never loads again after that all the content is loaded dynamically

[the remark this answers is not in the document]
Comment (ethanaobrien) [reply public/js/editor.js:11]: Maybe we should have the editor page be loaded not as a single page app, or in an iframe

**Review by ethanaobrien — APPROVED** [review]

**Pull request merged**